src/DynamoPackages/PackageLoader.cs

```diff
@@ -30,13 +31,31 @@ public class PackageLoader : LogSourceBase
         private readonly List<Package> localPackages = new List<Package>();
         public IEnumerable<Package> LocalPackages { get { return localPackages; } }
 
-        public string RootPackagesDirectory { get; private set; }
+        private readonly List<string> packagesDirectories = new List<string>();
+        public string DefaultPackagesDirectory
+        {
+            get { return packagesDirectories[0]; }
+        }
 
         public PackageLoader(string overridePackageDirectory)
+            : this(new[] { overridePackageDirectory })
+        {
+        }
+
+        public PackageLoader(IEnumerable<string> packagesDirectories)
         {
-            RootPackagesDirectory = overridePackageDirectory;
-            if (!Directory.Exists(RootPackagesDirectory))
-                Directory.CreateDirectory(RootPackagesDirectory);
+            if (packagesDirectories == null || (!packagesDirectories.Any()))
+                throw new ArgumentNullException("packagesDirectories");
+
+            this.packagesDirectories.AddRange(packagesDirectories);
+            try
+            {
+                if (!Directory.Exists(DefaultPackagesDirectory))
+                    Directory.CreateDirectory(DefaultPackagesDirectory);
+            }
+            catch (IOException) { }
+            catch (ArgumentException) { }
+            catch (UnauthorizedAccessException) { }
```

Comment:
That's a good point. @nguyen-binh-minh, since you're pushing new code changes in here, can you try to fold these into your code changes? Thanks!

```csharp
Exception exception = null;

try
{
    if (!Directory.Exists(DefaultPackagesDirectory))
        Directory.CreateDirectory(DefaultPackagesDirectory);
}
catch (IOException e) { exception = e; }
catch (ArgumentException e) { exception = e; }
catch (UnauthorizedAccessException e) { exception = e; }

if (e != null)
    this.Log(e.Message);
```